R/tt_export.R

```diff
@@ -630,5 +427,517 @@ export_as_pdf <- function(tt,
+    indent_size <- matform$indent_size * 5
+  }
+  for (i in seq_len(NROW(tt))) {
+    flx <- flextable::padding(flx,
+      i = i, j = 1,
+      padding.left = indent_size * rdf$indent[[i]]
+    )
+  }
+
+  # Adding referantial footer line separator if present
+  if (length(matform$ref_footnotes) > 0) {
+    flx <- flextable::add_footer_lines(flx, values = matform$ref_footnotes) %>%
+      add_hborder(part = "body", ii = nrow(tt), border = border)
+  }
+
+  # Footer lines
+  if (length(all_footers(tt)) > 0) {
+    flx <- flextable::add_footer_lines(flx, values = all_footers(tt))
+  }
+
+  flx <- flextable::width(flx, width = final_cwidths)
+
+  if (!is.null(theme)) {
+    flx <- theme(flx)
+  }
+
+  # Title lines (after theme for problems with lines)
+  if (titles_as_header &&
+    length(all_titles(tt)) > 0 && any(nzchar(all_titles(tt)))) {
+    real_titles <- all_titles(tt)
+    real_titles <- real_titles[nzchar(real_titles)]
+    flx <- flextable::add_header_lines(flx, values = real_titles, top = TRUE) %>%
+      # Remove the added borders
+      remove_hborder(part = "header", w = c("inner", "top")) %>%
+      # Re-add the separator between titles and real headers
+      add_hborder(
+        part = "header", ii = length(real_titles),
+        border = border
+      ) %>%
+      # Remove vertical borders added by theme eventually
+      remove_vborder(part = "header", ii = seq_along(real_titles))
+  }
+  # These final formatting need to work with colwidths
+  flx <- flextable::set_table_properties(flx, layout = "autofit")
+  flx <- flextable::fix_border_issues(flx) # needed?`
+  flx
+}
+
+#' @describeIn tt_to_flextable main theme function for [export_as_docx()]
+#' @param font character(1). Defaults to `"Arial"`. If the font is not vailable, `flextable`
+#'   default is used.
+#' @param font_size integer(1). Positive integerish value that defaults to 9.
+#' @param bold character vector. It can be any combination of `c("header", "content_rows",
+#'   "label_rows")`. The first one renders all column names bold (not `topleft` content).
+#'   Second and third option use [rtables::make_row_df()] to render content or/and label
+#'   rows as bold.
+#' @param bold_manual named list. List of indexes lists. See example for needed structure.
+#'   Accepted groupings/names are `c("header", "body")`.
+#' @inheritParams export_as_docx
+#'
+#' @seealso [export_as_docx()]
+#'
+#' @examples
+#' # Custom theme
+#' special_bold <- list(
+#'   "header" = list("i" = 1, "j" = c(1, 3)),
+#'   "body" = list("i" = c(1, 2), "j" = 1)
+#' )
+#' custom_theme <- theme_docx_default(tbl,
+#'   font_size = 10,
+#'   font = "Brush Script MT",
+#'   border = officer::fp_border(color = "pink", width = 2),
+#'   bold = NULL,
+#'   bold_manual = special_bold
+#' )
+#' tt_to_flextable(tbl,
+#'   border = officer::fp_border(color = "pink", width = 2),
+#'   theme = custom_theme
+#' )
+#'
+#' @export
+theme_docx_default <- function(tt = NULL, # Option for more complicated stuff
```

Comment:
very nice!

Reply:
why do we need a tt here?

Reply:
it is because the requirements need the label rows to be bold so I retrieve that from `maker_row_df(tt)`. I can possibly use the bold_manual option that I have created but I like how simple and compact the character option for bold is

Reply:
i feel this theme should be independent from an rtables object, and this should be something that is user facing, let the user to provide the theme requirement, and if later on it is needed for other manipulation, it should be a seperate function to interact with the table object, would you agree?

Reply:
it makes sense! Will do it ;)